gate.lic

```diff
@@ -0,0 +1,194 @@
+custom_require.call(%w[common common-arcana events spellmonitor])
+
+# expects a 'gate' waggle set
+# expects raw channeling feat 
+
+class Moongate
+  include DRC
+  include DRCA
+
+  def initialize
+    arg_definitions = [
+      [
+        { name: 'person', regex: /\w+/, description: 'Person to gate' },
+        { name: 'release', regex: /release/i, optional: true, description: 'Release gate once person comes through' },
+        { name: 'teleport', regex: /teleport/i, optional: true, description: 'Cast teleport instead of moongate' }
+      ]
+    ]
+
+    args = parse_args(arg_definitions)
+    @settings = get_settings
+    @lnet = (Script.running + Script.hidden).find { |val| val.name == 'lnet' }
```

Comment:
I was working towards extracting LNet messaging into a common place but never quite finished. This would be the third place it's used (after bankbot and lichbot), so it might be worth finishing that effort...

Reply:
yeah I agree. im not really sure how lnet interaction works. I copied this out of bankbot/lichbot.